dynamiqs/propagator/sepropagator.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import jax.numpy as jnp
+from jaxtyping import ArrayLike
+
+from ..core.expm_solver import ExpmSolver
+from ..gradient import Gradient
+from ..options import Options
+from ..result import PropagatorResult, Saved
+from ..sesolve import sesolve
+from ..solver import Expm, Solver, Tsit5
+from ..time_array import ConstantTimeArray, PWCTimeArray, TimeArray
+from ..utils.operators import eye
+
+__all__ = ["propagator"]
+
+
+def propagator(
```

Comment:
Sorry: right that this doesn't work but I was wrong before about the reason. It is because of the module dynamiqs/sesolve/sepropagator.py. When I do 
```python
from dynamiqs import sepropagator
``` 
it is not clear whether I am importing the function `sepropagator` or the module (it likely depends on the order of the init file?). This name collision is unfortunate because we also have the module dynamiqs/propagator/sepropagator.py, but hopefully as you say the name stuff will be ironed out?

Reply:
Indeed this is very bad naming from our side I think. I have made an internal RFC on how to reorganize/rename these solvers. Hopefully this should be a PR soon and we'll be able to fix this!